ios-trivia-game/ViewControllers/FinalScoreViewController.swift

```diff
@@ -9,27 +9,112 @@
 import UIKit
 
 class FinalScoreViewController: UIViewController {
-
+    
+    @IBOutlet weak var timerButton: UIBarButtonItem!
+    @IBOutlet weak var finishButton: UIButton!
+    @IBOutlet weak var maxScoreLabel: UILabel!
+    @IBOutlet weak var finalScoreTable: UITableView!
+    
+    var roomId: String!
+    
+    fileprivate var timerCount = 60
+    fileprivate var countdownTimer = Timer()
+    
+    fileprivate var finalScores: [FinalScore] = []
+    fileprivate var currentUserFinalScore: FinalScore?
+    
     override func viewDidLoad() {
         super.viewDidLoad()
 
-        // Do any additional setup after loading the view.
+        self.countdownTimer = Timer.scheduledTimer(timeInterval: 1, target: self, selector: #selector(updateCounter), userInfo: nil, repeats: true)
+        
+        self.finalScoreTable.delegate = self
+        self.finalScoreTable.dataSource = self
+        self.finalScoreTable.estimatedRowHeight = 100
+        self.finalScoreTable.rowHeight = UITableViewAutomaticDimension
+        
+        // given a room id, find all players in the game
+        // for each player, create a FinalScore
+        FirebaseClient.instance.getUsersInGameBy(roomId: roomId, complete: { (snapshot) in
```

Comment:
If someone knows of a better way to implement these three successive async calls, i am open to suggestions.

Perhaps chaining???